R/geom_pointdensity.R

```diff
@@ -13,27 +13,67 @@ count_neighbors_r <- function(x, y, r2, xy) {
   })
 }
 
+
+#' Wraps the user supplied Geom (typically GeomPoint) to add class "check_aspect_grob" and information about the aspect ratio assumed under which the densities were calculated to the grobs it draws
+#' The check is injected by providing an S3 instance to the makeContext generic that is called by grid for each grob at render time (where the actual plot aspect ratio is finally known)
```

Comment:
Just out of curiosity, does this mean the density gets re-calculated each time the viewer pane in RStudio is resized?

Reply:
No. If you did not use either `+theme(aspect.ratio=``)` or `+coord_fixed()`, it will just emit a warning every time you resize the pane (unless you adjust it perfectly such that the actual panel aspect ratio is 1)